modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/tracing/opentelemetry/management/handling/span/SpanHandler.java

```diff
@@ -51,25 +53,79 @@
+public class SpanHandler implements OpenTelemetrySpanHandler {
 
-    private Log logger = LogFactory.getLog(JaegerSpanHandler.class);
+    private Log logger = LogFactory.getLog(SpanHandler.class);
     /**
      * The tracer object, that is used to hold all the Jaeger spans.
      */
-    private JaegerTracer tracer;
+    private Tracer tracer;
+    private static OpenTelemetry openTelemetry;
 
     /**
      * Manages tracing scopes.
      * Useful during cases like when an API is called within Proxy service.
      */
     private TracingScopeManager tracingScopeManager;
 
-    public JaegerSpanHandler(JaegerTracer tracer, TracingScopeManager tracingScopeManager) {
-        this.tracer = tracer;
+    public SpanHandler(TelemetryTracer tracer, OpenTelemetry openTelemetry, TracingScopeManager tracingScopeManager) {
+        this.tracer = tracer.getTelemetryTracingTracer();
+        SpanHandler.openTelemetry = openTelemetry;
         this.tracingScopeManager = tracingScopeManager;
     }
 
+    /**
+     * Inject tracer specific information to tracerSpecificCarrier.
+     *
+     * @param span                  Span which the span information will be injected to the tracerSpecificCarrier.
+     * @param tracerSpecificCarrier Hashmap to inject the tracer and span context.
+     */
+    public static void inject(io.opentelemetry.api.trace.Span span, Map<String, String> tracerSpecificCarrier) {
+
+        TextMapSetter<Map<String, String>> setter = new TextMapSetter<Map<String, String>>() {
+            @Override
+            public void set(Map<String, String> tracerSpecificCarrier, String key, String value) {
+
+                if (tracerSpecificCarrier != null) {
+                    tracerSpecificCarrier.put(key, value);
+                }
+
+            }
+        };
+        try (Scope ignored = (span).makeCurrent()) {
+            openTelemetry.getPropagators().getTextMapPropagator().inject(Context.current(), tracerSpecificCarrier
+                    , setter);
+        }
+    }
+
+    /**
+     * Extract tracer specific information from tracerSpecificCarrier and return the extracted context.
+     *
+     * @param tracerSpecificCarrier Hashmap to extract the tracer and span context.
+     * @return extracted context.
+     */
+    public static Context extract(Map<String, String> tracerSpecificCarrier) {
+
+        TextMapGetter<Map<String, String>> getter =
+                new TextMapGetter<Map<String, String>>() {
+                    public String get(Map<String, String> tracerSpecificCarrier, String key) {
+
+                        if (tracerSpecificCarrier != null && tracerSpecificCarrier.containsKey(key)) {
+                            return tracerSpecificCarrier.get(key);
+                        }
+                        return null;
+                    }
+
+                    public Iterable<String> keys(Map<String, String> tracerSpecificCarrier) {
+
+                        return tracerSpecificCarrier.keySet();
```

Comment:
Shouldn't we check whether `tracerSpecificCarrier` is null (since we are checking it in the above `get()` method, I hope there's a possibility that this can be `null`)?